linkerd/transport-header/src/lib.rs

```diff
@@ -269,3 +269,31 @@ mod tests {
         assert_eq!(&buf, b"12345");
     }
 }
+
+#[cfg(fuzzing)]
+pub mod fuzz_logic {
+    use super::*;
+    pub fn fuzz_entry(fuzz_data: &str, ran_data: &[u8]) {
+        if let Ok(name) = Name::from_str(fuzz_data) {
+            let _header = TransportHeader {
+                port: 4040,
+                name: Some(name),
+                protocol: Some(SessionProtocol::Http2),
```

Comment:
I agree - this is one my todo list, namely improving the fuzzers to have randomised data in essentially all inputs. However, can we set this as a todo rather than fix in this PR?